doc/ui-guidelines.md

```diff
@@ -0,0 +1,257 @@
+- `theme/with-theme` wrapper is not needed anymore, `(theme/use-theme-value)` hook can be used directly in the components
+- `:f>` not needed anymore, all components are functional by default
+- `rn/use-callback` hook should be used for anon callback functions
+
+> [!IMPORTANT]
+> DO NOT USE anon functions directly in the props
+ 
+BAD
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         #(set-pressed true)
+      :on-press-out        #(set-pressed nil)}]))
+```
+
+## Global state and subscriptions
+
+For global state management, we utilize Re-frame subscriptions. They can be likened to React state. To obtain the state, `(rf/sub [])` is employed, and to modify it, `(rf/dispatch [])` is utilized. However, they update components in a similar manner to React states.
+
+```clojure
+(defn view
+  [{:keys [selected-tab]}]
+  (let [collectible-list (rf/sub [:wallet/all-collectibles])
+        on-collectible-press (rn/use-callback
+                              (fn [{:keys [id]}]
+                               (rf/dispatch [:wallet/get-collectible-details id])))]
+    [rn/view {:style style/container}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view {:collectibles         collectible-list
+                                         :on-collectible-press on-collectible-press}])
+       [activity/view]]))
+```
+
+## Regular atoms
+
+In some cases regular atoms were used in the components
+
+
+BEFORE:
+```clojure
+(defn view
+  []
+  (let [focused? (atom false)]
+    (fn []
+      (let [on-clear               (fn []
+                                     (reset! status (if @focused? :active :default)))
+            on-focus               (fn []
+                                     (reset! focused? true))
+            on-blur                (fn []
+                                     (reset! focused? false))]))))
+```
+
+NOW:
+```clojure
+(defn view
+  []
+  (let [[focused? set-focused] (rn/use-state false)
+        on-clear               (rn/use-callback
+                                (fn []
+                                  (set-status (if focused? :active :default)))
+                                [focused?])
+        on-focus               (rn/use-callback
+                                (fn []
+                                  (set-focused true)))
+        on-blur                (rn/use-callback
+                                (fn []
+                                  (set-focused false)))]))
+```
+
+> [!IMPORTANT]
+> Instead of a regular atom, we also use a state hook. It's important to understand that its value is now encapsulated within the function, and we must pass it as a dependency to `use-callback` so that the function re-executes with the new value.
```

Comment:
this a really good point, i looked at this twice, and now i think i agree with you

Reply:
so @J-Son89 do you think we should still use atoms, because i don't see good way to useRef directly without atom, so 2 options we could just use `rn/use-ref(atom(default-value))` or something like `rn/use-atom(default-value)` edyt?

Reply:
interesting, I'd have to experiment a bit myself. Can you expand a bit on the issue of why we would need an atom for useRef?

Reply:
maybe its just something we can abstract away for devs too so it _feels_ like using a ref and they don't need to know about this issue? 🤔

Reply:
no, because there are no variables in clojure, and you have to use atoms for such cases,  so yes `rn/use-ref(atom(default-value))` looks like the best way to do it, so devs know that its a ref, and value is atom

Reply:
so then `rn/use-ref-atom(default-value)`

Reply:
```clojure
(defn use-ref-atom [value]
   (let [ref (use-ref(atom value))]
      (.current ^js ref)))
```

Reply:
or we could use it similarly to state, something like
```clojure
[value set-value] (rn/use-ref default-value)
```
but this might be confusing, idk

Reply:
cc @ulisesmac @ilmotta

Reply:
@J-Son89 i updated doc with `use-ref-atom` looks good to me